polybuild/polybuild.py

```diff
@@ -0,0 +1,345 @@
+        if track_list_files is None:
+            print("Error! Failed to add track_lists")
+            return False
+        for file in ignore_list_files:
+            opt_command.append("-polytrack-dfsan-abilist=" + file)
+        for file in track_list_files:
+            opt_command.append("-polytrack-dfsan-abilist=" + file)
+        opt_command += [input_file, "-o", bitcode_file]
+        ret_code = subprocess.call(opt_command)
+        if ret_code != 0:
+            print("Error! opt command failed!")
+            return False
+        if not os.path.exists(bitcode_file):
+            print("Error! Bitcode file does not exist!")
+            return False
+        return True
+
+    def poly_instrument(self, input_file, output_file, bitcode_file, libs) -> bool:
+        res = self.poly_opt(input_file, bitcode_file)
+        if not res:
+            print(f"Error instrumenting bitcode {input_file} with opt!")
+            return False
+        res = self.poly_compile(bitcode_file, output_file, libs)
+        if not res:
+            print(f"Error compiling bitcode!")
+            return False
+        return True
+
+    # TODO add qUnusedArgs here
+    def poly_build(self, argv) -> bool:
+        compile_command = []
+        if self.meta.is_cxx:
+            compile_command.append("gclang++")
+        else:
+            compile_command.append("gclang")
+        compile_command += ["-pie", "-fPIC"]
+        if self.meta.is_cxx:
+            compile_command.append("-stdlib=libc++")
+            compile_command.append("-nostdinc++")
+            compile_command.append("-I" + self.meta.compiler_dir + "/../cxx_libs/include/c++/v1/")
+            compile_command.append("-L" + self.meta.compiler_dir + "/../cxx_libs/lib/")
+        for arg in argv[1:]:
+            compile_command.append(arg)
+        is_linking = self.poly_is_linking(argv)
+        if is_linking:
+            # If its cxx, link in our c++ libs
+            if self.meta.is_cxx:
+                compile_command += ["-lc++", "-lc++abipoly", "-lc++abi", "-lpthread"]
+        res = subprocess.call(compile_command)
+        if res != 0:
+            return False
+        return True
+
+
+"""
+Store a build artifact to the artifact storage via copy
+"""
+
+
+def store_artifact(file_path, artifact_path) -> bool:
+    filename = get_file_name(file_path)
+    artifact_file_path = artifact_path + "/" + filename
+    # Check if its an absolute path
+    if file_path[0] != "/":
+        cwd = os.getcwd()
+        targ_path = cwd + "/" + file_path
+    else:
+        targ_path = file_path
+    if not os.path.exists(targ_path):
+        print(f"Error! cannot find {targ_path}")
+        return False
+    ret = subprocess.call(["cp", targ_path, artifact_file_path])
+    if ret != 0:
+        print(f"Error! failed to store {targ_path}, error was {ret}")
+        print(f"Artifact path: {artifact_file_path}")
+        return False
+    return True
+
+
+def get_file_name(file_path: str) -> str:
```

Comment:
Not exactly sure what this function is meant for, but you might be able to replace it with [`os.path.split`](https://docs.python.org/3/library/os.path.html#os.path.split) or [`os.path.basename`](https://docs.python.org/3/library/os.path.html#os.path.basename)

Reply:
I think you are correct, ill investigate